[CNSMR-1959][Draft] What I don't know, but should (#216)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/WhatIDontKnow.md

# What I don't know, but should

> ## Congratulations you’ve joined Babylon...

Reviewer: ```suggestion
## Congratulations! You've joined Babylon...
```

Welcome to the amazing iOS team! 

> Babylon is a fast-growing company as you could see we are onboarding ~ 30 people weekly many of them engineers like yourself. Our team is scattered across three offices: London West, London East and US office. On top of that, we have fully remote teammates. As you can imagine this poses almost biblical challenges, but don’t worry :)

Reviewer: ```suggestion
Babylon is a fast-growing company. Currently, we are on-boarding ~30 new-starters weekly. Many of them are engineers like yourself. Our team is scattered across three offices: London West, London East and US office. On top of that, we have fully remote teammates. As you can imagine this poses almost biblical challenges, but don’t worry :)
```

> Babylon is a fast-growing company. Currently, we are on-boarding ~30 new-starters weekly. Many of them are engineers like yourself. Our team is scattered across three offices: London West, London East and US office. On top of that, we have fully remote teammates. As you can imagine this poses almost biblical challenges, but don’t worry :)

Reviewer 2: ```suggestion
Babylon is a fast-growing company. Currently, we are on-boarding ~30 new-starters weekly. Many of them are engineers like yourself. Our iOS team is scattered across three offices: London West, London East, and the US office. On top of that, we have fully remote teammates. As you can imagine this poses almost biblical challenges, but don’t worry :)
```

We have more offices that have non-iOS people, so I added `iOS` to differentiate.

## Like others before you...

> The first week may be difficult, all of the reading and chatting make for a bit of a mind-numbing first experience. We don’t promise the information firehouse stops after the first day, but just hear what people who get through are saying:

Reviewer: ```suggestion
The first week can be overwhelming. New code-base, emails and non-stop messages on Slack can make for a mind-numbing first experience. We don’t promise the information fire hose stops after the first day, but just hear what people who went through before you said:
```

Q: What did you like during your first week?

> > “1-on-1 with Rui and squad PMs, meet team, talk a lot within the squad, pair with a couple of people on iOS team. Gone out to drinks with the squad :)”

Reviewer 3: Since Rui is no longer here, this could be confusing. 
```suggestion
> “1-on-1 with Rui _(line manager)_ and squad PMs, meet team, talk a lot within the squad, pair with a couple of people on iOS team. Gone out to drinks with the squad :)”
```

Reviewer 4: I am always here. 😄

> > “I found the team to be very helpful in the first while becoming familiar with the code ...”

Reviewer 3: ```suggestion
> “I found the team to be very helpful in the first week while becoming familiar with the code ...”
```

> “Working in my squad on real tickets from day one”

> “Quickly started working on an actual project, smooth onboarding because of the iOS playbook, people being extremely helpful and welcoming, 1st day with the team”

> > “Team welcoming both iOS and squad ”

Reviewer 5: [Typo] whitespace in quote.
```suggestion
> “Team welcoming both iOS and squad”
```

## You’ll never walk alone…
As you will notice, due to the rapid growth, things in Babylon can and will change often and quickly. While this may look scary, do not worry - you are working with amazing people and everyone wants to help you. You have the right to ask questions and seek out help from people in the team and other co-workers. To make this task easier, here’s an incomplete list of people that can help you.

### Your manager

> Your manager will be the first person from the iOS team that you’ll see after the induction workshop on your first day. He, or she, should introduce you to your buddy, your squad and other co-workers.

Reviewer 3: ```suggestion
Your manager will be the first person from the iOS team you’ll see after the induction workshop on your first day. They should introduce you to your buddy, your squad, and other co-workers.
```

- [iOS Onboarding Line Manager Checklist](https://docs.google.com/spreadsheets/d/1Cu_e1Ol4nkpDNawQOeCbho9xg5L02-1k-Y1YCZjBZ_c/edit#gid=0)

> Why you should read it: This will tell you how your first 3 months at Babylon should look like.

Reviewer 3: [nit] I'd add some formatting to `Why you should read it` everywhere

Author: great I will add it

### Your buddy

> When you start, you will be paired up with a colleague - who has some work experience in Babylon

Reviewer 3: ```suggestion
When you start, you will be paired up with a colleague who has some work experience in Babylon
```

Think of this buddy as your mentor, as the go-to person if you have any questions. They should answer technical questions, introduce the team, and even walk you through some code or pair programming.

- [iOS Buddy Onboarding Checklist](https://docs.google.com/spreadsheets/d/1nKxNwAWdscPjIDOqrKpyU7zefF1fBuad0STXe2R0WiA/edit#gid=0)

Why you should read it: This will give you a good idea what you can expect during your onboarding.

> How to be a good buddy?

Author: @chitrakotwani would you see "buddy" section expanded, or just moved to the bottom?

> How to be a good buddy?

Reviewer: Maybe just move it to the end. Like - 
```
# Good buddy
Things to bear in mind when you become a buddy in due course :) 
- Point 1
- Point 2
```

> How to be a good buddy?

Reviewer 6: I also had a 'Babylon buddy' - someone who is not in the iOS - and we went for a coffee. But I've heard from other people they never met theirs, I'm not sure it's worth mentioning.

>Treat it as you would introduce a friend, who is new to your city: Introduce them to locals, give them some resources so they can navigate on their own, and help them form a plan their first few weeks.

### Your squad

> You may be still confused by the name, but a squad is a cross-functional, autonomous team working towards common goals. Get to know them well during your onboarding time. They will be more than happy to explain and help you with day-to-day work. Check what are your squad slack channels, calendar groups and JIRA board and make sure that you are added to them.

Reviewer 2: ```suggestion
You may be still confused by the name, but a squad is a cross-functional, autonomous team working towards common goals. Think of it as a mini-startup. Get to know them well during your onboarding time. They will be more than happy to explain and help you with day-to-day work. Check what are your squad slack channels, calendar groups and JIRA board and make sure that you are added to them.
```

### The iOS team

> One of the greatest sources of our collective knowledge is a #ios-questions slack channel. You can ask there any question and the whole team will try to help you.

Reviewer: ```suggestion
One of the greatest sources of our collective knowledge is #ios-questions slack channel. You should search for your question there before asking it. We encourage to ask your questions on this channel as opposed to asking it in private chat.
```

> One of the greatest sources of our collective knowledge is #ios-questions slack channel. You should search for your question there before asking it. We encourage to ask your questions on this channel as opposed to asking it in private chat.

Reviewer 7: Worth mentioning `#ios-underground` and `#ios-pedia`.

## In the tower of Babel

> When working in such a vibrant and dynamic environment it is very easy to lose a track of things that should happen. This is a shortlist of topics that should help you to get through the onboarding time.

Reviewer 2: ```suggestion
When working in such a vibrant and dynamic environment it is very easy to lose track of things that should happen. This is a shortlist of topics that should help you to get through the onboarding time.
```

### Communication and Collaboration
Effective and open communication is crucial in a company like Babylon. Without it, everything falls apart. If there is one thing you should remember from this guide let me suggest this:
Context is king!

When you’re communicating digitally, you never quite know what the other person is doing at that moment. They might be at their desk just like you are, or they may be frantically rushing to a meeting, only responding “Yes” to your question and not elaborating because they don’t have time.

Without understanding the other person’s context, you might think that person doesn’t care about the issue you brought up when they’re really just running to a meeting room. With the information on that person’s context, all of a sudden, the curt answers make sense: It’s not that your coworker doesn’t care, they are just indisposed at the moment.

> Prefacing communication with your context can help to prevent any miscommunication when things are out of the ordinary. Let team members know when you are heads down on a project and can’t respond to questions right away.

Reviewer 5: 👍 very cool to have this suggested explicitly.

> Over communicating is always better than making assumptions.

Reviewer 2: This this this. Great.

[Remote team communication](https://zapier.com/learn/remote-work/remote-team-communication/)

Why you should read it: This will give you a bunch of tips on how to communicate better digitally.

[Remote communication](https://about.gitlab.com/2016/03/23/remote-communication/index.html)

Why you should read it: It has some universal commandments about communicating asynchronously (between humans ;) )

[Communication: how to be a better software developer](https://medium.com/techspiration-ideas-making-it-happen/communication-how-to-be-a-better-software-developer-869c50767701)

Why you should read it: A bit longer read, but presents good practices we all should strive to achieve.

### Am I missing something?
This is our FAQ section. Every question that you may have have a short answer and/or links to additional resources. If during your onboarding you ever find yourself asking a question that is not on this list, create a new item, find an answer with your buddy and update the board.

#### List of events you should have in your calendar
- Your squad’s daily/retro/ planning
- Babylon standup
- Frontend demo
- Proposal meetings
- PR party
- Lunch and Learn
- iOS Banter

> - 1-on-1 with your manager

Reviewer 6: What do you think about adding a small description for  the meetings?
I was a bit lost what was happening, especially when I joined my first banter or got invited to the PR party.

Author: That's a great idea. Will add that

Reviewer 6: Had some time today :) I was thinking something like this, but feel free to modify. I haven't added any formatting either, maybe the explanation could be italic for better readability?

- Your squad’s daily/retro/planning: ask your product or delivery manager how your squad organises these and make sure they invited you to all of them.
- Babylon standup: this weekly standup helps us keep up with all things happening in the different teams in Babylon.
- Frontend demo: curious what's going on in other Frontend teams? Participate at Frontend demo and have a glimpse of their work.
- Proposal meetings: Have a suggestion about how we can improve our way of work? Create a proposal for it so we can exchange ideas about it. You can find our proposals in the Cookbook.
- PR party: Seen your name group up with your teammates and a bunch of PR numbers in the ios-underground channel? Congratulations you got invited to your first PR party!
The first person on the list will slack you and invite you to a Zoom call where you will review all the assigned pull requests.
- Lunch and Learn: grab your lunch and join the team to watch a video or listen to someone's presentation.
- iOS Banter: we gather together every week to share our concerns and successes.
- 1-on-1 with your manager: biweekly catch up with your manager where you can discuss anything, they are to help you grow to be the best you can be.

Reviewer 7: You are missing the iOS Retro meeting.
Maybe you could link these to the document that describes the chapter meetings https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/meetings-purpose.md

> - Frontend demo

Reviewer 3: No longer needed
```suggestion
```

Reviewer 3: It's back :D

#### List of documents you should star on the Google drive

> [iOS Team Plan End 2018 - 2019](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit#gid=0)

Reviewer: ```suggestion
[iOS Team Plan, Dec 2018 - present](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit#gid=0)
```

Author: @chitrakotwani this is the document tile unfortunately and I'd prefer to keep it as it is to avoid confusion.

> #### List of documents you should star on the Google drive

Reviewer 3: I'd remove links from this section, they change periodically and would become outdated unless updated here. Let's keep the title generic and no links

> Why you should read it: This document shows Who, What When in our team

Reviewer: ```suggestion
Why you should read it: In this document, we keep track of who, what and when of our team
```

> [iOS Goals H2 (July - December)](https://docs.google.com/spreadsheets/d/1nI1RAdsP2nJiUzsvCE9LHQH11-OIPNSHZvpMPUQal_o/edit#gid=1533898989)

Reviewer: ```suggestion
[iOS Goals H2 (July - December)](https://docs.google.com/spreadsheets/d/1nI1RAdsP2nJiUzsvCE9LHQH11-OIPNSHZvpMPUQal_o/edit#gid=0)
```

> Why you should read it: Here you can see our goals not related to daily squad work.

Reviewer: ```suggestion
Why you should read it: Here, the iOS team collaborates on it's goals (unrelated to squad goals). Typically, these goals include addressing tech debts, refactoring the code for modernisation and implementing features which make development more efficient and more robust
```

> [iOS Goals H2 (July - December)](https://docs.google.com/spreadsheets/d/1nI1RAdsP2nJiUzsvCE9LHQH11-OIPNSHZvpMPUQal_o/edit#gid=0)

Reviewer 3: Are the goals tracked here or on Jira? /cc @AnnKatF

Reviewer 7: @sdaniel55 in both places 🙂

### Jargon translator

> In our team’s daily conversations very often you will encounter strangely looking/sounding terms. This section is to help you get familiar with both our technical jargon and cultural references. Only here you can learn what does :renderable: mean on the Slack :)

Reviewer: ```suggestion
In our team’s daily conversations very often you will encounter strangely looking/sounding terms. This section is to help you get familiar with both our technical jargon and cultural references. Only here you can learn what :renderable: means on the Slack :)
```

> Why you should read it: Here, the iOS team collaborates on it's goals (unrelated to squad goals). Typically, these goals include addressing tech debts, refactoring the code for modernisation and implementing features which make development more efficient and more robust

Reviewer 3: ```suggestion
Why you should read it: Here, the iOS team collaborates on its goals (unrelated to squad goals). Typically, these goals include addressing tech debts, refactoring the code for modernisation and implementing features which make development more efficient and more robust
```

#### What is NVL?

This is our New Visual Language, more at: [NVL](https://babylonpartners.atlassian.net/wiki/spaces/V4/pages/946077746/NVL+Project+doc)

Why you should read it: You can’t escape from the NVL so it’s better to know what it is about.

#### What is Octopus?

[Octopus 🐙](https://babylonhealth.slack.com/archives/GD6J0QWQ5/p1548151935005200)

Why you should read it: Maybe not aloud, but absolutely everyone asked himself that question.

#### What are WOWs?

What, Owner, When, more at:
[WOWs](https://sites.google.com/babylonhealth.com/induction/about-us/wows)

Why you should read it: This is something that will show up in your plannings quite often.

> #### What is :renderable: emoji?

Reviewer 3: Is this still relevant? I have never seen this 👀

It escalated from [this](https://babylonhealth.slack.com/archives/G9S9L0TEK/p1557410403027900)
to [this](https://babylonhealth.slack.com/archives/G22EWKXQT/p1557428026144000)

Why you should read it: Wait till you see it on the slack :D

### Reading list
This is the list of docs, wikis, confluence pages and articles that we think you should read to get the context of how and what we are doing.

[iOS Cookbook](https://github.com/Babylonpartners/ios-playbook/blob/master/Cookbook/README.md)

Why you should read it: This is our “Bible”, nothing more to add.

[iOS Cookbook - style guide](https://github.com/Babylonpartners/ios-playbook/tree/master/Cookbook/Style-guide)

Why you should read it: Worth to get familiar with before making your first PR.

[iOS Cookbook - technical documents](https://github.com/Babylonpartners/ios-playbook/tree/master/Cookbook/Technical-Documents)

Why you should read it: Those may answer many of your technical questions.

> 

Reviewer 3: Maybe add the "[New Hires Checklist](https://github.com/Babylonpartners/ios-playbook/blob/master/Cookbook/Technical-Documents/NewHiresCheckList.md)" as well? It has useful lists for accounts and Slack channels, and the guide to set up dev environment

[Playbook PRs](https://github.com/Babylonpartners/ios-playbook/pulls)

Why you should read it: Those are things that are, better to be up to date.

[Squads, tribes, chapters…](https://www.growly.io/spotify-engineering-model-with-squads-tribes-chapters-and-guilds/)

Why you should read it: Admit, all those squads, tribes, chapters are confusing from time to time. Here you can make more sense of it.

[... and some more](https://medium.com/@media_75624/exploring-key-elements-of-spotifys-agile-scaling-model-471d2a23d7ea)

Why you should read it: If the first article wasn’t enough this may be helpful.

[Babylon squads, tribes, chapters](https://sites.google.com/babylonhealth.com/induction/about-us/structure)

Why you should read it: The chart at the bottom should show you the whole picture.

[Pointfree](https://www.pointfree.co/)

> Why you should read it: This is almost a Bible to us when it comes to functional programming in Swift.

Reviewer: ```suggestion
Why you should read it: Our codebase is heavily inspired by the concepts there. If you go through the lectures, you will understand much better what we do in our codebase and why.
```

## Life in Babylon
Those resources contain practical tips about living in our offices or as a remote worker. Floors, toilets, conference rooms, lunch places and so on. 

[New starter induction guide](https://sites.google.com/babylonhealth.com/induction/home)

Why you should read it: This is a lot to ingest but you will get a lot of useful info about daily life at Babylon.

[Babylon Heart](https://heart.ops.babylontech.co.uk/)

Why you should read it: Our mini-portal where you can find a bunch of useful links.
Note: This is only available from our internal network or through the VPN

[Engineering at Babylon](https://engineering.ops.babylontech.co.uk/)

> Why you should read it: It provides a nice overview of the engineering across the whole Babylon.

Reviewer: ```suggestion
Why you should read it: It provides a nice overview of engineering across whole Babylon.
```

[Remote Working at Babylon](https://docs.google.com/document/d/1hjKtx1pPL_pxhLEuW-46w2XwxsC-mkw3Cp5uaHB2P8c/edit#)

> Why you should read it: If you work remotely this contains a bunch of useful info.

Reviewer: ```suggestion
Why you should read it: If you work remotely, this contains a bunch of useful tips.
```
